beacon/beacon.go

```diff
@@ -154,3 +177,35 @@ func (f forwardOrigin) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 	}
 	f(w, req)
 }
+
+func (api *Server) newBQClient(req *http.Request, projectID string) (*bigquery.Client, error) {
+	switch api.AuthMode {
+	case ServiceAuth:
+		return bigquery.NewClient(appengine.NewContext(req), projectID)
+	case UserAuth:
+		return newClientFromBearerToken(req.WithContext(appengine.NewContext(req)), projectID)
+	default:
+		panic(fmt.Sprintf("invalid value %d for server authentication mode", api.AuthMode))
```

Comment:
Return errors wherever possible instead of panicing

The panic in the other case is OK because you don't really have another option.